tests/plugins/test_pulp_pull.py

```diff
@@ -299,13 +299,17 @@ def test_pull_push_vs_sync(self, push, sync):
         assert results == test_id
         assert len(tasker.pulled_images) == 1
 
-    @pytest.mark.parametrize('v2', [False, True])
+    @pytest.mark.parametrize('v2,expect_v2schema2', [
+        (False, False),
+        (True, False),
+        (True, True),
+    ])
     @pytest.mark.parametrize('timeout,retry_delay,failures,expect_success', [
         (0.1, 0.06, 1, True),
         (0.1, 0.06, 1, True),
         (0.1, 0.06, 3, False),
     ])
-    def test_pull_retry(self, v2, timeout, retry_delay, failures,
+    def test_pull_retry(self, expect_v2schema2, v2, timeout, retry_delay, failures,
```

Comment:
This method tests how the plugin reacts to 404 responses. Would be great to have another one that tests how it reacts to v2schema1 responses when expect_v2schema2 is True.

Reply:
It does happen for any `v2 == True` - first a number of 404's returned, then a schema 1manifest in returned (`expectation.and_return(self.config_response_config_v1)`) and after that a schema 2 manifest is returned

Reply:
Not sufficient. If I undo the relevant post_pulp_pull.py changes, the tests still pass.

Reply:
Oh, indeed. Added a warning log message and a test which is expected to fail due to the timeout, as pulp never returns schema 2, but its expected